### PR TITLE
Improve build instructions for libcu++

### PR DESCRIPTION
**Pull request by miscco**
Our build isntructions are heavily outdated, this tries to improve the situation a bit

### CODE REVIEW

File: docs/libcudacxx/setup/building_and_testing.rst [hunk 1]
```diff
@@ -27,176 +21,116 @@ In a Bash shell:
...
-   export LIBCUDACXX_ROOT=/path/to/libcudacxx # Git repo root.
-
-Step 1: Generate the Build Files
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-In a Bash shell:
-
-.. code:: bash
+Windows, Native Build/Test
+===========================
```
Comment (bernhardmgruber) [inline docs/libcudacxx/setup/building_and_testing.rst:25]: Q: Are those instructions for native build/test on Windows (no linux), or for Windows and any native build/test on other OSs? Maybe to can rephrase the heading to make this more clear.

File: docs/libcudacxx/setup/building_and_testing.rst [hunk 2]
```diff
@@ -27,176 +21,116 @@ In a Bash shell:
...
-\*nix Systems, NVRTC Build/Test
--------------------------------
+.. code:: bash
 
-The procedure is demonstrated for NVRTC in C++11 mode on a Debian-like
-Linux systems; the same basic steps are required on all other platforms.
+   ./ci/windows/build_libcudacxx.ps1
 
-.. _step-0-install-build-prerequisites-1:
+This should cover most users needs and can be adopted to work with different inputs:
```
Comment (bernhardmgruber) [inline docs/libcudacxx/setup/building_and_testing.rst:56]:
```suggestion
This should cover most users' needs and can be adapted to work with different inputs:
```

File: docs/libcudacxx/setup/building_and_testing.rst [hunk 3]
```diff
@@ -27,176 +21,116 @@ In a Bash shell:
...
 
-Step 0: Install Build Prerequisites
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. code:: bash
 
-Follow Step 0 for \*nix native builds/tests.
+    -cxx  Selects the host compiler
+    -std  Selects the C++ standard mode
+    -cuda Selects the cuda compiler
```
Comment (bernhardmgruber) [inline docs/libcudacxx/setup/building_and_testing.rst:62]: Q: isn't CUDA spelled ALLCAPS?
```suggestion
    -cuda Selects the CUDA compiler
```

File: docs/libcudacxx/setup/building_and_testing.rst [hunk 4]
```diff
@@ -27,176 +21,116 @@ In a Bash shell:
...
+    -arch Selects the cuda architectures
 
-.. _step-1-generate-the-build-files-2:
+Optionally one can directly invoke cmake
```
Comment (bernhardmgruber) [inline docs/libcudacxx/setup/building_and_testing.rst:65]:
```suggestion
Optionally, one can directly invoke cmake
```

File: docs/libcudacxx/setup/building_and_testing.rst [hunk 5]
```diff
@@ -27,176 +21,116 @@ In a Bash shell:
...
 
-Step 1: Generate the Build Files
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Optionally Step 2: Manually generate the Build Files:
```
Comment (bernhardmgruber) [inline docs/libcudacxx/setup/building_and_testing.rst:67]:
```suggestion
Optional Step 2: Manually generate the Build Files:
```
Applies to more places.

File: docs/libcudacxx/setup/building_and_testing.rst [hunk 6]
```diff
@@ -27,176 +21,116 @@ In a Bash shell:
...
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If only building the tests is desired one can pass ``-Dexecutor="NoopExecutor()"`` to the lit invocation.
+.. code:: bash
 
-In a Visual Studio command prompt:
+   cd build
+   lit libcudacxx-cpp17/RELATIVE_PATH_TO_TEST_OR_SUBFOLDER -sv -Dexecutor="NoopExecutor()"
 
-.. code:: bat
+Finally different standard modes can be tested by passing e.g ``--param=std=c++20``
```
Comment (bernhardmgruber) [inline docs/libcudacxx/setup/building_and_testing.rst:118]:
```suggestion
Finally, different standard modes can be tested by passing e.g ``--param=std=c++20``
```

File: docs/libcudacxx/setup/building_and_testing.rst [hunk 7]
```diff
@@ -27,176 +21,116 @@ In a Bash shell:
...
 
-   cmake ^
-     -S ./ ^
-     -B build ^
-     -G "Ninja" ^
-     -DCMAKE_CXX_COMPILER=cl ^
-     -DCMAKE_CUDA_COMPILER=nvcc ^
-     -DCMAKE_CUDA_COMPILER_FORCED=ON ^
-     -DLIBCUDACXX_ENABLE_LIBCUDACXX_TESTS=ON
+NVRTC tests can be build and tested the same way as the other tests
```
Comment (bernhardmgruber) [inline docs/libcudacxx/setup/building_and_testing.rst:123]:
```suggestion
NVRTC tests can be built and tested the same way as the other tests
```

File: docs/libcudacxx/setup/building_and_testing.rst [hunk 8]
```diff
@@ -27,176 +21,116 @@ In a Bash shell:
...
+.. code:: bash
 
-Step 2: Build & Run the Tests
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   cmake --preset libcudacxx-nvrtc-cpp17
+   cmake --build --preset libcudacxx-cpp17
+   ctest --preset libcudacxx-lit-cpp17
 
-``SM_ARCH`` can be set to any integer value (Ex: “80”, “86”)
+If you want to run individual tests its again
```
Comment (bernhardmgruber) [inline docs/libcudacxx/setup/building_and_testing.rst:131]: Suggestion: Sounds a bit sloppy, maybe:
```suggestion
If you want to run individual tests, you can just run again:
```

**Review by ericniebler — APPROVED** [review]
this is AWESOME thank you @miscco. i agree with @bernhardmgruber's suggestions. other than that, looks good. it's a big improvement.

File: docs/libcudacxx/setup/building_and_testing.rst
Comment (jrhemstad) [inline docs/libcudacxx/setup/building_and_testing.rst:1]: suggestion: A lot of the stuff that is in here is redundant with what is in the top-level docs here: https://github.com/NVIDIA/cccl/blob/main/CONTRIBUTING.md#building-and-testing

I'd prefer to consolidate towards unified docs like these. 

I do recognize the need and value for having specific instructions for how to use `lit`.

I'd propose deleting this entire file and instead starting a "libcudacxx Developer Guide" document similar to the CUB one's here: https://github.com/NVIDIA/cccl/blob/main/CONTRIBUTING.md#cub

That should include a section on using `lit` directly to invoke tests.

Comment (bernhardmgruber) [reply docs/libcudacxx/setup/building_and_testing.rst:1]: Let's follow up with this later: #2935

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**